pkg/api/meta/builds.go

```diff
@@ -0,0 +1,55 @@
+package meta
+
+import (
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+)
+
+type buildSpecMutator struct {
+	spec *buildapi.CommonSpec
+	path *field.Path
+}
+
+func (m *buildSpecMutator) Mutate(fn ImageReferenceMutateFunc) field.ErrorList {
+	var errs field.ErrorList
+	for i, image := range m.spec.Source.Images {
+		if image.From.Kind != "DockerImage" {
```

Comment:
When build configs are triggered they resolve their ImageStreamTag/Images down to DockerImages - so normally builds are created with images.  I think it's valid to do an extra resolution here, although I'll need to change to a better way to handle that.